File: utils/removetaskpnts.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
# vi: set ft=python sts=4 ts=4 sw=4 et:


from argparse import (ArgumentParser, RawTextHelpFormatter)
import numpy as np
import pandas as pd

#AZEEZ

def get_parser():

    parser = ArgumentParser(
        formatter_class=RawTextHelpFormatter,
        description='Remove non steady volumes '
    )
    parser.add_argument(
        '-n', '--nvd', action='store', required=True,default=0,
        help=' number of line to be deleted')
    parser.add_argument(
        '-t', '--tab', action='store', required=True,
        help='[required]'
             '\n confound regressors table')
    parser.add_argument(
        '-s', '--sab', action='store', required=True,
        help='[required]'
             '\n return confound regressors table')

    return parser


opts = get_parser().parse_args()

nvd=opts.nvd
tab = pd.read_csv(opts.tab, sep='\t')
tad = tab.loc[:, 1:nvd]
newtab.to_csv(opts.sab, encoding='utf-8', index=False, sep='\t')

File: utils/removetaskpnts.R
#! /usr/bin/env Rscript

################################################################### 
#  ⊗  ⊗  ⊗  ⊗  ⊗  ⊗  ⊗  ⊗  ⊗  ⊗  ⊗  ⊗  ⊗  ⊗  ⊗  ⊗  ⊗  ⊗  ⊗  ⊗  ⊗  #
###################################################################

###################################################################
# You are advised to avoid excessive dependency on this script
# for the time being.
#~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~#

###################################################################

###################################################################
# Load required libraries
###################################################################
suppressMessages(suppressWarnings(library(optparse)))
suppressMessages(suppressWarnings(library(pracma)))

###################################################################
# Parse arguments to script, and ensure that the required arguments
# have been passed.
###################################################################
option_list = list(
   make_option(c("-t", "--tab"), action="store", default='null', type='character',
              help=" regresssor from fmriprep to check if there any steady volume             
                  etc."),
   make_option(c("-n", "--nvd"), action="store", default=0, type='integer',
              help="number of columns to remove "),
    make_option(c("-p", "--put"), action="store", default=NA, type='character',
              help="print out the table")
)
opt = parse_args(OptionParser(option_list=option_list))

if (is.na(opt$put)) {
   cat('User did not specify an output path.\n')
   cat('Use removenonsteady.R -h for an expanded usage menu.\n')
   quit()
}


nvd <- opt$nvd
tab1 <- opt$tab
out2 <- opt$put

rr=read.table(tab1, sep='\t',header=TRUE)
mat1 = rr[-c(1:nvd),] 
write.table(mat1,file=out2, sep='\t',col.names = colnames(rr), row.names=F,quote=F)



